### DISCUSSION TITLE
What is the 4th request cache middleware provided by Giraffe?

### DISCUSSION THREAD

**Discussion by 64J0** [giraffe-fsharp/Giraffe#551] [Q&A] [answered]
When reading the documentation at [Giraffe docs](https://giraffe.wiki/docs#response-caching), there's this part of the text:

> Giraffe offers in total 4 http handlers which can be used to configure response caching for an endpoint.

But, at this section I was able to locate only 3 http handlers:

* `publicResponseCaching`
* `privateResponseCaching`
* `noResponseCaching`

Then, what is the 4th handler?

**Comment by 64J0** [comment, marked as answer]
When looking at the code I found this function: `responseCaching`. But I'm not sure if it's the real 4th handler mentioned in the text.

From here -> https://github.com/giraffe-fsharp/Giraffe/blob/master/src/Giraffe/ResponseCaching.fs#L42.

**Reply by TheAngryByrd** [reply]
Looks like it is the 4th handler. It's the base function that all the provided helpers are created from, gives you absolute control but isn't as friendly.

**Reply by 64J0** [reply]
Indeed, it's presented in the next section `VaryByQueryKeys`. I did not notice it before.
